elog ordering panic

The following Edit command or a file `test.md`

```
Edit {
	,x/five/ c/10/
	e test.md
}
```

works (albeit with strange undo behaviour) in Acme. Conversely in Edwood, it's sad:

```
panic: Changes not in order

goroutine 68 [running]:
main.(*Elog).Delete(0xc000116648, 0x0, 0x599)
	/Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/elog.go:183 +0x28c
main.e_cmd(0xc000248928, 0xc0000b43c0, 0x200)
	/Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/ecmd.go:248 +0x25f
main.cmdexec(0xc000248928, 0xc0000b43c0, 0x1)
	/Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/ecmd.go:107 +0x379
main.cmdexec(0xc000248928, 0xc0000b43c0, 0x0)
	/Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/ecmd.go:101 +0x233
main.editthread()
	/Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/edit.go:131 +0x42
created by main.editcmd
	/Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/edit.go:210 +0x19d
```

Got something similar with an attempt at a more complicated `Edit` op:
```
Untested: main.run(0xc0003c8a80, 0xc0002a78f0, 0x4, 0xc000af0700, 0x3a, 0x1, 0x0, 0x0, 0x0, 0x0, ...): /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/exec.go:667 +0x34
panic: Changes not in order

goroutine 1112 [running]:
main.(*Elog).Insert(0xc000b54828, 0x27e6, 0xc00028abf0, 0x3, 0x20)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/elog.go:151 +0x4b0
main.appendx(0xc000b547e0, 0xc000358780, 0x27e6, 0xc0007e6150)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/ecmd.go:664 +0xdc
main.i_cmd(0xc0003c8b88, 0xc000358780, 0x27e6)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/ecmd.go:294 +0x45
main.cmdexec(0xc0003c8b88, 0xc000358780, 0x10)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/ecmd.go:108 +0x37a
main.loopcmd(0xc000b547e0, 0xc000358780, 0xc0004e75d0, 0x1, 0x1)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/ecmd.go:712 +0x89
main.looper(0xc000b547e0, 0xc000358730, 0xc0007e6001)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/ecmd.go:753 +0x1bc
main.x_cmd(0xc0003c8b88, 0xc000358730, 0x27eb)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/ecmd.go:451 +0x49
main.cmdexec(0xc0003c8b88, 0xc000358730, 0x116ed01)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/ecmd.go:108 +0x37a
main.cmdexec(0xc0003c8b88, 0xc000358730, 0xc0003585f0)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/ecmd.go:102 +0x230
main.editthread(0xc0001abcc0)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/edit.go:153 +0xbd
created by main.editcmd
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/edit.go:223 +0x1be

--- a/acme.go
+++ b/acme.go
@@ -469,6 +469,7 @@ func keyboardthread(display draw.Display) {
 					activecol = t.col
 				}
 				if t != nil && t.w != nil {
+					// In a set of zeroxes, the last typed-in body becomes the curtext.
 					t.w.body.file.curtext = &t.w.body
 				}
 				if timer != nil {

--- a/edit_test.go
+++ b/edit_test.go
@@ -363,6 +363,27 @@ func TestEditMultipleWindows(t *testing.T) {
 		{Range{0, 0}, fn3, "w", []string{contents, alt_contents}, []string{
 			fn3 + " not written; file already exists\n",
 		}},
+
+		// b
+		{Range{0, 0}, "test", "b alt_example_2\ni/inserted/\n", []string{
+			contents,
+			"inserted" + alt_contents,
+		}, []string{
+			"'+  alt_example_2\n",
+		}},
+		{Range{0, 0}, "test", "b alt_example_2\n1 i/1/\n2 i/2/\n", []string{
+			contents,
+			"1A different text\n2With other contents\nSo there!\n",
+		}, []string{
+			"'+  alt_example_2\n",
+		}},
+		// TODO(rjk): the edit result here is wrong. See #236.
+		{Range{0, 0}, "test", "b alt_example_2\n2 i/2/\n1 i/1/\n", []string{
+			contents,
+			"1A different text2\nWith other contents\nSo there!\n",
+		}, []string{
+			"'+  alt_example_2\nwarning: changes out of sequence\nwarning: changes out of sequence, edit result probably wrong\n",
+		}},
 	}
 
 	buf := make([]rune, 8192)

--- a/elog.go
+++ b/elog.go
@@ -10,7 +10,8 @@ const (
 	DeleteType ElogType = iota
 	InsertType
 	FilenameType
-	Wsequence = "warning: changes out of sequence\n"
+	Wsequence     = "warning: changes out of sequence\n"
+	WsequenceDire = "warning: changes out of sequence, edit result probably wrong\n"
 )
 
 // Elog is a log of changes made by editing commands.  Three reasons for this:
@@ -114,7 +115,8 @@ func (e *Elog) Replace(q0, q1 int, r []rune) {
 	eo.nd = q1 - q0
 	eo.setr(r)
 	if eo.q0 < e.secondlast().q0 {
-		panic("Changes not in order")
+		e.warned = true
+		warning(nil, WsequenceDire)
 	}
 }
 
@@ -148,7 +150,8 @@ func (e *Elog) Insert(q0 int, r []rune) {
 	eo.setr(r)
 
 	if eo.q0 < e.secondlast().q0 {
-		panic("Changes not in order")
+		e.warned = true
+		warning(nil, WsequenceDire)
 	}
 }
 
@@ -178,7 +181,8 @@ func (e *Elog) Delete(q0, q1 int) {
 	eo.q0 = q0
 	eo.nd = q1 - q0
 	if eo.q0 < e.secondlast().q0 {
-		panic("Changes not in order")
+		e.warned = true
+		warning(nil, WsequenceDire)
 	}
 }
 

--- a/row.go
+++ b/row.go
@@ -278,6 +278,7 @@ func (row *Row) Type(r rune, p image.Point) *Text {
 	if t != nil && !(t.what == Tag && p.In(t.scrollr)) {
 		w = t.w
 		if w == nil {
+			// Texts in column tags or the very top.
 			t.Type(r)
 		} else {
 			w.Lock('K')

